app/bundles/CoreBundle/Tests/Unit/Helper/ExportHelperTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/*
+ * @copyright   2020 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\CoreBundle\Tests\Unit\Helper;
+
+use Mautic\CoreBundle\Helper\ExportHelper;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class ExportHelperTest extends \PHPUnit\Framework\TestCase
+{
+    /** @var TranslatorInterface */
+    private $translatorInterfaceMock;
+
+    private $dummyData = [
+        [
+            'id'        => 1,
+            'firstname' => 'Mautibot',
+            'lastname'  => 'Mautic',
+            'email'     => 'mautibot@mautic.org',
+        ],
+        [
+            'id'        => 2,
+            'firstname' => 'Demo',
+            'lastname'  => 'Mautic',
+            'email'     => 'demo@mautic.org',
+        ],
+    ];
+
+    protected function setUp(): void
+    {
+        $this->translatorInterfaceMock = $this->createMock(TranslatorInterface::class);
+    }
+
+    /**
+     * Test if exportDataAs() correctly generates a CSV file when we input some array data.
+     */
+    public function testCsvExport()
+    {
+        $helper = $this->getExportHelper();
+        $stream = $helper->exportDataAs($this->dummyData, ExportHelper::EXPORT_TYPE_CSV, 'demo-file.csv');
+
+        $this->assertInstanceOf(StreamedResponse::class, $stream);
+        $this->assertSame(200, $stream->getStatusCode());
+        $this->assertSame(false, $stream->isEmpty());
+
+        ob_start();
+        $stream->sendContent();
+        $content = ob_get_contents();
+        ob_end_clean();
+
+        $csvData = $this->parseCsvString($this->removeBomUtf8($content));
+
+        $this->assertSame('1', $csvData[0]['id']);
+        $this->assertSame('Mautibot', $csvData[0]['firstname']);
+        $this->assertSame('2', $csvData[1]['id']);
+        $this->assertSame('Demo', $csvData[1]['firstname']);
+    }
+
+    /**
+     * Test if exportDataAs() correctly generates an Excel file when we input some array data.
+     */
+    public function testExcelExport()
+    {
+        $helper = $this->getExportHelper();
+        $stream = $helper->exportDataAs($this->dummyData, ExportHelper::EXPORT_TYPE_EXCEL, 'demo-file.xlsx');
+
+        $this->assertInstanceOf(StreamedResponse::class, $stream);
+        $this->assertSame(200, $stream->getStatusCode());
+        $this->assertSame(false, $stream->isEmpty());
+
+        ob_start();
+        $stream->sendContent();
+        $content = ob_get_contents();
+        ob_end_clean();
+
+        // We need to write to a temp file as PhpSpreadsheet can only read from files
+        file_put_contents('./demo-file.xlsx', $content);
```

Comment:
I'm not quite sure if the spreadsheet content can be previewed as a string. I imagine it's some type of XML. But I'd rather assert that than converting it to an object and asserting the values.

Reply:
Excel files are actually binary files (a type of ZIP IIRC), so it's rather difficult to test against the raw content I think 😉 I'll just leave it as-is for now, but your suggestion definitely makes sense to limit the scope of the test more.

![image](https://user-images.githubusercontent.com/17739158/99888336-57605600-2c4c-11eb-9aca-e00fc48024e3.png)